src/mountinfo.rs

```diff
@@ -0,0 +1,462 @@
+                    if is_comment_line(row) {
+                        return None;
+                    } else {
+                        return Some(parse_mount_point(row, self.row_num));
+                    }
+                },
+            }
+        }
+    }
+}
+
+fn parse_mount_point(row: &[u8], row_num: usize)
+     -> Result<MountPoint, ParseError> 
+{
+    let invalid_format = || {
+        ParseError(format!("Expected more fields"), row_num, 
+            String::from_utf8_lossy(row).into_owned())
+    };
+
+    let mut cols = row.split(|c| *c == b' ');
+    let mount_id = try!(parse_int(&mut cols, row, row_num));
+    let parent_id = try!(parse_int(&mut cols, row, row_num));
+    let mut major_minor = try!(cols.next().ok_or_else(&invalid_format))
+        .split(|c| *c == b':');
+    let major = try!(parse_int(&mut major_minor, row, row_num));
+    let minor = try!(parse_int(&mut major_minor, row, row_num));
+    let root = try!(parse_path(&mut cols, row, row_num));
+    let mount_point = try!(parse_path(&mut cols, row, row_num));
+    let mount_options = try!(parse_os_str(&mut cols, row, row_num));
+    let mut optional_fields = vec!();
+    let mut opt_field = try!(parse_os_str(&mut cols, row, row_num));
+    while opt_field != OsStr::new("-") {
+        optional_fields.push(opt_field.clone());
+        opt_field = try!(parse_os_str(&mut cols, row, row_num));
+    }
+    let fstype = try!(parse_os_str(&mut cols, &row, row_num));
+    let mount_source = try!(parse_os_str(&mut cols, &row, row_num));
+    let super_options = try!(parse_os_str(&mut cols, &row, row_num));
+    Ok(MountPoint {
+        mount_id: mount_id,
+        parent_id: parent_id,
+        major: major,
+        minor: minor,
+        root: root,
+        mount_point: mount_point,
+        mount_options: mount_options,
+        optional_fields: optional_fields,
+        fstype: fstype,
+        mount_source: mount_source,
+        super_options: super_options,
+    })
+}
+
+fn is_comment_line(row: &[u8]) -> bool {
+    if row.is_empty() {
+        return true;
+    }
+    for c in row {
+        if *c == b' ' || *c == b'\t' {
+            continue;
+        }
+        if *c == b'#' {
+            return true;
+        }
+        return false;
+    }
+    return false;
+}
+
+fn strip_line(row: &[u8]) -> &[u8] {
+    let mut row = row;
+    while row.ends_with(&[b'\r']) {
+        row = &row[..row.len() - 1]
+    }
+    row
+}
+
+fn parse_os_str(columns: &mut Iterator<Item=&[u8]>, row: &[u8], row_num: usize)
+    -> Result<OsString, ParseError>
+{
+    let bytes = try!(columns.next()
+        .ok_or_else(|| ParseError(
+            format!("Expected more fields"),
+            row_num, String::from_utf8_lossy(row).into_owned())));
+    let mut value = Cow::Borrowed(bytes);
+    unescape_octals(&mut value);
+    Ok(OsString::from_vec(value.into_owned()))
+}
+
+fn parse_int(columns: &mut Iterator<Item=&[u8]>, row: &[u8], row_num: usize)
```

Comment:
You don't need your own int parser. Just use `std::str::from_utf8` and fail immediately when data is not utf-8.